protocols/relay/src/v2/relay.rs

```diff
@@ -757,8 +757,12 @@ impl Action {
                     addrs: poll_parameters
                         .external_addresses()
                         .map(|a| {
-                            a.addr
-                                .with(Protocol::P2p((*poll_parameters.local_peer_id()).into()))
+                            let p2p_proto =
+                                Protocol::P2p(*poll_parameters.local_peer_id().as_ref());
+                            match a.addr.iter().last() {
+                                Some(p) if p == p2p_proto => a.addr,
```

Comment:
I don't think we need to check whether the `PeerId` is our own. This check should happen earlier in the process, in case it is not already happening.